File: Python/0136_single_number/solution.py
def single_number(nums):
    num_count = {}
    for num in nums:
        if num not in num_count:
            num_count[num] = 1
        else:
            num_count[num] += 1

    for key,value in num_count.items():
        if value == 1:
            return key
    return

nums = [4,1,2,1,2]
print(single_number(nums))

File: JavaScript/0136_single_number/solution.js
function single_number(nums) {
    let num_count = {};
    for (let num of nums) {
        num_count[num] = (num_count[num] || 0) + 1;
    }

    for (let [key,value] of Object.entries(num_count)) {
        if (value == 1) {
            return Number(key);
        }
    }
    return null;
}

const nums = [4,1,2,1,2];
console.log(single_number(nums));